src/psyclone/nemo.py

```diff
@@ -57,6 +57,20 @@ class NemoFparser2Reader(Fparser2Reader):
     '''
     Specialisation of Fparser2Reader for the Nemo API.
     '''
+    @staticmethod
+    def _create_schedule(_, invoke):
```

Comment:
Overloading `create_schedule` to deal with kernels and invokes seems like shoehorning to me. If feasible I think it would make more sense for there to be a `create_kernel_schedule`, `create_invoke_schedule` and eventually we would add `create_algorithm_schedule`. Whether these would belong in fparser2reader or whether they would be PSyclone specialisations of fparser2reader I am not sure. However, my current thinking is that fparser2reader should not know about PSyKAl - it just reads Fortran and parses it into PSyIR so would lean toward specialisations. Perhaps you, I and @sergisiso could discuss this.

Reply:
@arporter @rupertford I see the problem now. For other APIs create_schedule means create a kernelSchedule because is the only contexts it is called (for now), while for Nemo it means create InvokeSchedule because it is the only use (kernels are inlined).

I think removing the `invoke` parameter is the easy part, it can be done with a setter at line 184:
```
processor = NemoFparser2Reader()
self._schedule = processor.generate_schedule(name, ast)
self._schedule.invoke = self
```

I agree having fparser2Reader create Kernels/Invokes seems not right (and I chose the name very poorly). I think generate_~~schedule~~subroutine() should create 'Suboutines' instead (a new node we need) and the code calling the it decide if it is something more specific (like an Invoke or a Kernel). But this should be another PR because there is probably more subtleties. I would leave the current implementation of (without `invoke`) a todo referencing this issue.

Reply:
Unfortunately I can't use a setter for the `invoke` parameter because I actually need it within the `generate_schedule()` routine (in order to identify whether there's an associated `Container`). This shows that `generate_schedule()` should not be creating a `Container` - that should be done first and then the result passed in to `create_subroutine()` or somesuch.

Reply:
I've had quite a long think about this and decided that although I could create a new `generate_invoke_schedule` in the `nemo` module, there would be an awful lot of code duplication. Since we've agreed that this needs to be improved more generally, I've created #737 and added various TODOs. Thanks to Sergi's suggestion I have improved things slightly as I realised it was the presence (or otherwise) of an existing Container that I actually needed to know about within the `generate_schedule` routine - this is now passed explicitly. When we do #737 the creation of a Container should be handled separately (rather than being done as required in `generate_schedule`).